### PR TITLE
BaSP-MR-50: Admin Add/Edit Form

### PR DESCRIPTION
**Pull request by AleMac17**
Add route on Layout Component

Create add/edit form including its respective data (picture its just an example)

On edit functionality should show previous info

Success message for confirmation edit and add actions

Warining/error messages for edit and add

### CODE REVIEW

**Comment by AleMac17** [comment]
[[Admin] Add/Edit Form](https://trello.com/c/F4stn3ZS/60-admin-add-edit-form)

File: src/Components/Admins/Form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from 'react';
+import styles from './form.module.css';
+
+const Form = ({ postAdminForm, putAdminForm, bodyEdit }) => {
+  const [firstName, setName] = useState('');
```
Comment (Gonzarey96) [inline src/Components/Admins/Form/index.jsx]:
```suggestion
  const [firstName, setFirstName] = useState('');
```

File: src/Components/Admins/Form/index.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      email,
+      city,
+      password
+    };
+    if (bodyEdit.length !== 0) {
+      putAdminForm(formData);
+    } else {
+      postAdminForm(formData);
+    }
+    /* bodyEdit.length !== 0 ? putAdminForm(formData) : postAdminForm(formData);*/
```
Comment (Gonzarey96) [inline src/Components/Admins/Form/index.jsx]: Delete this comment

File: src/Components/Admins/index.jsx [hunk 3]
```diff
@@ -37,48 +42,117 @@ const Admins = () => {
...
             deleteAdmins={deleteAdmins}
-            title="Warning"
-            body="Do you want to delete this admin?"
+            title={modalInformation.title}
+            body={modalInformation.body}
           />
           <h2>Admins</h2>
-          <button className={styles.newButton} href="#">
+          <button className={styles.newButton} onClick={() => setShowForm(true)}>
             + Add new Admin
```
Comment (Gonzarey96) [inline src/Components/Admins/index.jsx]: There are 2 buttons to add a new Admin, this was the idea?

File: src/Components/Admins/index.jsx [hunk 4]
```diff
@@ -37,48 +42,117 @@ const Admins = () => {
...
-          <Table data={admins} deleteItem={deleteItem} />
+          <Table data={admins} deleteItem={deleteItem} handleEdit={handleEdit} />
         </>
       ) : (
         <>
           <h2>Admins</h2>
           <h3>There are no admins in the database</h3>
-          <button className={styles.newButton} href="#">
+          <button className={styles.newButton} onClick={() => setShowForm(true)}>
             + Add new Admin
```
Comment (Gonzarey96) [inline src/Components/Admins/index.jsx]: There are 2 buttons to add a new Admin, this was the idea?

**Review by Gonzarey96 — APPROVED** [review]
In general the code seems well written and follows good practices, it is readable and the use of descriptive variables is appreciated, congratulations!

File: src/Components/Classes/index.jsx [hunk 5]
```diff
@@ -11,6 +11,7 @@ const Classes = () => {
     try {
       const response = await fetch(` ${process.env.REACT_APP_API_URL + '/classes'}`);
       const { data } = await response.json();
+      console.log(data);
```
Comment (fernandofernandezuy) [inline src/Components/Classes/index.jsx]: You should remove this console.log()

**Review by LucianoAlarcon — APPROVED** [review]

**Pull request merged**